Content.Server/Health/BodySystem/Surgery/SurgeryData/BiologicalSurgeryData.cs

```diff
@@ -94,30 +139,68 @@ protected void CautizerizeIncisionSurgery(BodyManagerComponent target, IEntity p
             _vesselsClamped = false;
             _skinRetracted = false;
         }
-        protected void DisconnectOrganSurgery(BodyManagerComponent target, IEntity performer)
+
+
+        protected void LoosenOrganSurgery(IBodyPartContainer container, ISurgeon surgeon, IEntity performer)
+        {
+            if (_parent.Mechanisms.Count <= 0)
+                return;
+            List<Mechanism> toSend = new List<Mechanism>();
+            foreach (Mechanism mechanism in _parent.Mechanisms)
+            {
+                if (!_disconnectedOrgans.Contains(mechanism))
+                    toSend.Add(mechanism);
+            }
+            if (toSend.Count > 0)
+                surgeon.RequestMechanism(toSend, LoosenOrganSurgeryCallback);
+        }
+        public void LoosenOrganSurgeryCallback(Mechanism target, IBodyPartContainer container, ISurgeon surgeon, IEntity performer)
         {
-            Mechanism mechanismTarget = null;
-            //TODO: figureout popup, right now it just takes the first organ available if there is one
-            if (_parent.Mechanisms.Count > 0)
-                mechanismTarget = _parent.Mechanisms[0];
-            if (mechanismTarget != null)
+            if (target != null && _parent.Mechanisms.Contains(target))
             {
                 ILocalizationManager localizationManager = IoCManager.Resolve<ILocalizationManager>();
```

Comment:
there's a statically available `Loc` reference to the `LocalizationManager`, prefer that.

Reply:
Fixed